File: app/src/main/java/com/platdmit/simplecloudmanager/app/fragments/ServerTabMainFragment.java
package com.platdmit.simplecloudmanager.app.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.platdmit.simplecloudmanager.R;
import com.platdmit.simplecloudmanager.app.adapters.LoadAverageListAdapter;
import com.platdmit.simplecloudmanager.app.vm.LoadAverageViewModel;
import com.platdmit.simplecloudmanager.app.vm.factory.SingleElementViewModelFactory;
import com.platdmit.simplecloudmanager.data.api.implement.ApiServerRepoImp;
import com.platdmit.simplecloudmanager.domain.SCMApp;
import com.platdmit.simplecloudmanager.domain.converters.implement.LoadAverageConvertImp;
import com.platdmit.simplecloudmanager.domain.helpers.ContentUpdateService;
import com.platdmit.simplecloudmanager.domain.models.LoadAverage;
import com.platdmit.simplecloudmanager.domain.repo.ServerLoadAveragesRepo;
import com.platdmit.simplecloudmanager.domain.repo.implement.ServerRepoImp;
import com.platdmit.simplecloudmanager.domain.repo.implement.UpdateScheduleRepImp;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

public class ServerTabMainFragment extends Fragment implements ServerTabFragment<ServerTabMainFragment>{
    private static final String TAG = ServerTabMainFragment.class.getSimpleName();
    private String mTitle;

    private LoadAverageViewModel mLoadAverageViewModel;
    private RecyclerView mLoadAverageRecyclerView;
    private LoadAverageListAdapter mLoadAverageListAdapter;

    public ServerTabMainFragment(){};

    public ServerTabMainFragment(String title) {
        mTitle = title;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        if(savedInstanceState != null){
            mLoadAverageViewModel = new ViewModelProvider(this).get(LoadAverageViewModel.class);
        } else {
            mLoadAverageViewModel = new ViewModelProvider(this,
                    new SingleElementViewModelFactory<ServerLoadAveragesRepo>(
                            new ServerRepoImp(
                                    new ApiServerRepoImp(SCMApp.getActualApiKeyService()),
                                    SCMApp.getDb(),
                                    new LoadAverageConvertImp(),
                                    new ContentUpdateService(new UpdateScheduleRepImp(SCMApp.getDb()))
                            ), ServerLoadAveragesRepo.class, getArguments().getLong("ELEMENT_ID")
                    )).get(LoadAverageViewModel.class);
        }

        View v = inflater.inflate(R.layout.fragment_server_tab_main, container, false);

        mLoadAverageRecyclerView = v.findViewById(R.id.server_load_average);
        mLoadAverageRecyclerView.setLayoutManager(new LinearLayoutManager(getContext()));

        mLoadAverageListAdapter = new LoadAverageListAdapter();

        mLoadAverageViewModel.getLoadAveragesLiveData().observe(getViewLifecycleOwner(), this::updateServerActionData);

        return v;
    }

    @Override
    public String getTitle() {
        return mTitle;
    }

    @Override
    public ServerTabMainFragment getInstance() {
        return new ServerTabMainFragment();
    }

    private void updateServerActionData(List<LoadAverage> actions){
        mLoadAverageListAdapter.setContentData(actions);
        if(mLoadAverageRecyclerView.getAdapter() == null){
            mLoadAverageRecyclerView.setAdapter(mLoadAverageListAdapter);
        } else {
            mLoadAverageListAdapter.notifyDataSetChanged();
        }
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/fragments/ServerTabMainFragment.kt
package com.platdmit.simplecloudmanager.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import com.platdmit.simplecloudmanager.R
import com.platdmit.simplecloudmanager.adapters.LoadAverageListAdapter
import com.platdmit.simplecloudmanager.vm.LoadAverageViewModel
import com.platdmit.domain.models.LoadAverage
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.android.synthetic.main.fragment_server_tab_main.*

@AndroidEntryPoint
class ServerTabMainFragment(
        private val mTitle: String = "empty"
) : Fragment(R.layout.fragment_server_tab_main), ServerTabFragment<ServerTabMainFragment> {
    private val loadAverageViewModel: LoadAverageViewModel by viewModels()
    private val loadAverageListAdapter = LoadAverageListAdapter()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if(savedInstanceState == null){
            loadAverageViewModel.setActiveId(requireArguments().getLong("ELEMENT_ID"))
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        server_load_average.layoutManager = LinearLayoutManager(context)
        loadAverageViewModel.loadAveragesLiveData.observe(viewLifecycleOwner, Observer { updateServerActionData(it) })
    }

    override fun getTitle(): String {
        return mTitle
    }

    override fun getInstance(): ServerTabMainFragment {
        return ServerTabMainFragment()
    }

    private fun updateServerActionData(actions: List<LoadAverage>) {
        loadAverageListAdapter.setContentData(actions)
        if (server_load_average.adapter == null) {
            server_load_average.adapter = loadAverageListAdapter
        } else {
            loadAverageListAdapter.notifyDataSetChanged()
        }
    }
}